### PR TITLE
Standardise csv names

### PR DESCRIPTION
**Pull request by SamJamCul**
## Changes proposed in this pull request
https://trello.com/c/PXP4BsEb/4323-standardise-csv-names-in-manage

Adding a csv naming helper, which improves consistency. 
There's still a bit of inconsistency based on what string is being passed in, maybe that could be put into the locales? 
The way different export controllers get the relevant year range is a bit inconsistent, so the helper is quite blunt in how it handles it - as long as you give it the recruitment cycle year/s, it works out a good date range string.

Does it need more tests? The helper has tests, and the controllers are all pretty much doing the same thing as before!

## Guidance to review

Logging in as a provider, head to the reports section. Try downloading each of the different available reports, and check the name of the downloaded file (and that it's a csv)

image

image

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Review by richardpattinson — APPROVED** [review]
Looks pretty uncontroversial to me, excellent job! =)

File: spec/helpers/csv_name_helper_spec.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+RSpec.describe CSVNameHelper, type: :helper do
+  around do |example|
+    Timecop.freeze(2020, 6, 1, 12) do
+      example.run
+    end
+  end
+
+  let(:export_name) { 'extremely-cool-data' }
+  let(:cycle_years) { [0, 2000] }
```
Comment (despo) [inline spec/helpers/csv_name_helper_spec.rb:11]: Shouldnt this be [2000] only?

File: app/helpers/csv_name_helper.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+module CSVNameHelper
+  def csv_filename(export_name:, cycle_years:, providers:)
+    "#{export_name}_#{year_range(cycle_years)}_#{provider_string(providers)}_#{Time.zone.now.strftime('%Y-%m-%d_%k-%M-%S')}.csv"
+  end
+
+private
+
+  def year_range(cycle_years)
+    sorted_cycle_years = cycle_years.sort - [0]
```
Comment (despo) [inline app/helpers/csv_name_helper.rb:9]: same question as below, why do we need `-[0]`?

File: spec/helpers/csv_name_helper_spec.rb (hunk 1 above)
Comment (SamJamCul) [reply spec/helpers/csv_name_helper_spec.rb:11]: For some reason the application data export adds a 0 to the cycle years array - this was a quick fix, but I'll do a bit of debugging to see why it's doing this 
image

**Pull request merged**